capability-providers/keyvalue/rust/src/main.rs

```diff
@@ -49,14 +53,46 @@ pub struct Invocation {
     pub origin: WasmCloudEntity,
     pub target: WasmCloudEntity,
     pub operation: String,
-    // may not need this since the host doesn't use it in the Rust invocation
-    //#[serde(with = "serde_bytes")]
+    // I believe we determined this is necessary to properly round trip the "bytes"
+    // type with Elixir so it doesn't treat it as a "list of u8s"
+    #[serde(with = "serde_bytes")]
     pub msg: Vec<u8>,
     pub id: String,
     pub encoded_claims: String,
     pub host_id: String,
 }
 
+/// The response to an invocation
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default)]
+pub struct InvocationResponse {
+    // I believe we determined this is necessary to properly round trip the "bytes"
+    // type with Elixir so it doesn't treat it as a "list of u8s"
+    #[serde(with = "serde_bytes")]
+    pub msg: Vec<u8>,
+    pub error: Option<String>,
+    pub invocation_id: String,
+}
+
+impl InvocationResponse {
+    pub fn failure(inv: &Invocation, e: &str) -> InvocationResponse {
+        InvocationResponse {
+            error: Some(e.to_string()),
+            invocation_id: inv.id.to_string(),
+            ..Default::default()
+        }
+    }
+
+    /// Creates a successful invocation response. All invocation responses contain the
+    /// invocation ID to which they correlate
+    pub fn success(msg: impl Serialize) -> InvocationResponse {
+        InvocationResponse {
+            invocation_id: "".to_string(), // to be filled in later,
```

Comment:
I prefer all of my intentionally empty values to be something like `YEET` or `HEYOO`, but this works too.

Reply:
having a const called `YEET` would let you write `YEET.into()`

Reply:
This is why I got into computer science.